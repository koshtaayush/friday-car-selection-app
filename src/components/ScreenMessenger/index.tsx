import React from 'react';
import styled from 'styled-components';
import error from './../../assets/error.svg'

interface Props {
    primaryMessage: string
    secondaryMessage: string
}

const ScreenMessenger: React.FC<Props> = (props) => {

    const { primaryMessage, secondaryMessage } = props
    return (
        <React.Fragment>
            <Container>
                <Image src={error} />
                <Messages>
                    <PrimaryMessage>{primaryMessage}</PrimaryMessage>
                    <SecondaryMessage>{secondaryMessage}</SecondaryMessage>
                </Messages>
            </Container>
        </React.Fragment>
    )
}

export default ScreenMessenger

const Container  = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    height: calc(100vh - 180px);
`

const Image = styled.img``

const Messages  = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
`

const PrimaryMessage  = styled.div`
    margin: 1rem;
    font-weight: bold;
`

const SecondaryMessage  = styled.div`
    color: #AAABAD
`