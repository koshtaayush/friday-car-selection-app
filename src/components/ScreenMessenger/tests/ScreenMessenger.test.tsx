import React from 'react'
import { shallow } from 'enzyme'

import ScreenMessenger from '..'

describe('ScreenMessenger', () => {
    it('Should match the snapshot', () => {

        const component = shallow(
            <ScreenMessenger
                primaryMessage={'Dummy Primary Message'}
                secondaryMessage={'Dummy Seconday Message'}
            />
        )

        expect(component).toMatchSnapshot()
    })

    it('Should render the ScreenMessenger text correctly', () => {

        const component = shallow(
            <ScreenMessenger
                primaryMessage={'Dummy Primary Message'}
                secondaryMessage={'Dummy Seconday Message'}
            />
        )

        expect(component.find({ 'test-id': 'primaryMessage' }).text()).toEqual('Dummy Primary Message')
        expect(component.find({ 'test-id': 'secondaryMessage' }).text()).toEqual('Dummy Seconday Message')
    })
    
})
