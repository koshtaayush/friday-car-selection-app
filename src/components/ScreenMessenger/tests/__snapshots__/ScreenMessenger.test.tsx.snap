// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenMessenger Should match the snapshot 1`] = `
<Fragment>
  <styled.div>
    <styled.img
      src="error.svg"
    />
    <styled.div>
      <styled.div
        test-id="primaryMessage"
      >
        Dummy Primary Message
      </styled.div>
      <styled.div
        test-id="secondaryMessage"
      >
        Dummy Seconday Message
      </styled.div>
    </styled.div>
  </styled.div>
</Fragment>
`;
